modules/apps/29-fee/ibc_middleware.go

```diff
@@ -293,10 +309,26 @@ func (im IBCMiddleware) OnTimeoutPacket(
 
 	packetID := channeltypes.NewPacketId(packet.SourcePort, packet.SourceChannel, packet.Sequence)
 	feesInEscrow, found := im.keeper.GetFeesInEscrow(ctx, packetID)
-	if found {
+	if !found {
+		// call underlying callback
+		return im.app.OnTimeoutPacket(ctx, packet, relayer)
+	}
+
+	payee, found := im.keeper.GetPayeeAddress(ctx, relayer.String(), packet.SourceChannel)
+	if !found {
 		im.keeper.DistributePacketFeesOnTimeout(ctx, relayer, feesInEscrow.PacketFees, packetID)
+
+		// call underlying callback
+		return im.app.OnTimeoutPacket(ctx, packet, relayer)
 	}
 
+	payeeAddr, err := sdk.AccAddressFromBech32(payee)
+	if err != nil {
+		return sdkerrors.Wrapf(err, "failed to create sdk.Address from payee: %s", payee)
+	}
```

Comment:
No! It should be impossible to set a bad address because in `ValidateBasic` of `MsgRegisterPayeeRequest`  we error if it fails `sdk.AccAddressFromBech32`. 

See here: https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/types/msgs.go#L48

In the event that this somehow becomes malformed it will error and can be reset by invoking the `RegisterPayee` tx again. This does perhaps come as a place to maybe `panic` but there is basically no difference and I would rather handle the error and bubble up.

Reply:
Ah okay makes sense, yeah I think just returning the error here makes sense in that case!